auslib/blobs/base.py

```diff
@@ -88,17 +46,28 @@ def createBlob(data):
 
 
 class Blob(dict):
-    """See isValidBlob for details on how format is used to validate blobs."""
-    format_ = {}
+    jsonschema = None
 
     def __init__(self, *args, **kwargs):
         self.log = logging.getLogger(self.__class__.__name__)
+        self.cached_schemas = {}
         dict.__init__(self, *args, **kwargs)
 
     def isValid(self):
         """Decides whether or not this blob is valid based."""
         self.log.debug('Validating blob %s' % self)
-        return isValidBlob(self.format_, self)
+        schema = self.getSchema()
+        # raises exception if broken
+        jsonschema.validate(self, schema)
+        return True
+
+    def getSchema(self):
+        if self.jsonschema in self.cached_schemas:
+            return self.cached_schemas[self.jsonschema]
+
+        schema = yaml.load(open(path.join(path.dirname(path.abspath(__file__)), "schemas", self.jsonschema)))
+        self.cached_schemas[self.jsonschema] = self.cached_schemas
```

Comment:
Yeah, this caching was busted in multiple ways. I fixed it to use the actual global cache in my latest commits.